string-cache-codegen/lib.rs

```diff
@@ -181,6 +181,25 @@ impl AtomType {
         )
     }
 
+    /// Write generated code to destination [`Vec<u8>`] and return it as [`String`]
+    /// 
+    /// Used mostly for testing or displaying a value. 
+    pub fn write_to_string(&mut self, mut destination: Vec<u8>) -> io::Result<String>
```

Comment:
Maybe this could be annotated with ` #[cfg(test)]`. I think it isn't useful otherwise and this can help to avoid breaking API in the future.

Reply:
I annotated it, but it caused breakage in the test.  So I converted tests to unit test.